### DISCUSSION TITLE
How does licensing work for subdeps with different licenses?

### DISCUSSION THREAD

**Discussion by wahidshafique** [gatsbyjs/gatsby#35250] [Help] [answered]
I noticed that [gatsby-plugin-sharp](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/package.json) uses potrace which is under [GPL 2.0](https://github.com/tooolbox/node-potrace/blob/master/LICENSE). However, the sharp plugin is listed as [MIT](https://www.npmjs.com/package/gatsby-plugin-sharp). I thought that the inclusion of potrace would in-turn make sharp the same license?

Via the Github explainer for GPL:
> When distributing derived works, the source code of the work must be made available under the same license. 

Can anyone shed some light on this, including whether use of sharp would present issues with licensing?

**Comment by wahidshafique** [comment]
Relevant: https://stackoverflow.com/questions/3902754/mit-vs-gpl-license

**Comment by jonmadison-amzn** [comment]
i am currently blocked from using gatsby-plugin-image, gatsby-plugin-sharp due to the GPL license of potrace. Have you (or anyone) ever gotten a usable update to this?

**Comment by LekoArts** [comment, marked as answer]
This is no longer the case, with https://github.com/gatsbyjs/gatsby/pull/35328 Gatsby uses [@gatsbyjs/potrace](https://github.com/gatsbyjs/node-potrace) with GNU 3 License to be compatible with MIT.

**Reply by selinger** [reply]
Hi, I am the copyright holder of Potrace, and a copyright holder of node-potrace, which is a derivative work of Potrace. I state unambiguously that the software at https://github.com/gatsbyjs/node-potrace is in violation of my copyright, since in a commit on 2022-03-29, the license has been changed from the GPL (GNU General Public License) to the LGPL (GNU Lesser General Public License), which is a different and incompatible license.

The licensing terms of Potrace are clear: Potrace and its derivative works can only be distributed under the terms of the GPL. Anybody who distributes Potrace under the terms of the LGPL or of the MIT license is in violation.

I just contacted @wardpeet and asked for this to be fixed. But I am also stating it here for the public record, to give notice to anybody who might have redistributed this software under an incorrect license that it is unauthorized and illegal to do so.

Thanks, -- Peter Selinger
